### PR TITLE
netdev_ieee802154: Add and use common reset function

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Refactors the common part from the radio reset functions to a netdev function. 

### Issues/PRs references

None

### Other

When netdev is layered, the `netdev_ieee802154_reset` function can be called from netdev_ieee802154 in response to a `NETOPT_STATE_RESET` set call and removed from the device drivers.

### CODE REVIEW

File: drivers/netdev_ieee802154/netdev_ieee802154.c [hunk 1]
```diff
@@ -50,6 +50,19 @@ static int _get_iid(netdev_ieee802154_t *dev, eui64_t *value, size_t max_len)
     return sizeof(eui64_t);
 }
 
+void netdev_ieee802154_reset(netdev_ieee802154_t *dev)
+{
+    dev->seq = 0;
```
Comment (miri64) [inline drivers/netdev_ieee802154/netdev_ieee802154.c]: Maybe use a random number here instead?

Comment (bergzand) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: I don't disagree with you, but that would be a change in behaviour for the radio drivers here. Would you mind if I do that as a small follow up to keep this PR purely as cleanup?

**Comment by bergzand** [comment]
I just noticed that the cast I'm doing could be a bit cleaner by passing `&dev->netdev` instead.

File: drivers/netdev_ieee802154/netdev_ieee802154.c (hunk 1 above)
Comment (miri64) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: :+1:

**Comment by bergzand** [comment]
> I just noticed that the cast I'm doing could be a bit cleaner by passing &dev->netdev instead.

Done

**Comment by PeterKietzmann** [comment]
Please squash the fixup commits

**Comment by bergzand** [comment]
Squashed!

**Comment by PeterKietzmann** [comment]
@miri64 would you remove your change request?

File: drivers/netdev_ieee802154/netdev_ieee802154.c (hunk 1 above)
Comment (smlng) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: I haven't found any usage of `seq` in one of the touched drivers, its used in the higher gnrc layers, but they are driver independent already. Hence, there should be no effect on the driver behaviour right now - or did I miss something?

Comment (bergzand) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: @smlng No, you're right, this should not affect the drivers. 

In the future I want to move the full `seq` handling to `netdev_ieee802154`.

Comment (bergzand) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: It is specific for ieee802154 (matching ACKs to frames), so I'm afraid moving it to `netdev` doesn't make sense here.

Comment (smlng) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: as said: just a question - I'm fine with keeping it .15.4 specific.

**Comment by smlng** [comment]
@miri64 good to go?

File: drivers/netdev_ieee802154/netdev_ieee802154.c (hunk 1 above)
Comment (smlng) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: btw. what do you mean by moving the `full handling of seq` to netdev_ieee802154? You mean the gnrc stuff?

Comment (bergzand) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: The problem I'm facing at the moment with implementing #7736 is that gnrc_netif_ieee802154 directly accesses data from the netdev struct. The problem with this is that gnrc_netif_ieee802154 assumes what kind of netdev device/layer is used. As long as this direct access is there, it is not possible to use anything else than a netdev_iee802154_t layer for the netif->dev. One possible solution for this is to move the handling of the `seq` number to the netdev_ieee802154 code.

**Comment by miri64** [comment]
> @miri64's comments were addressed in #9535.

Sorry, still working through a 2 day backlog of GitHub notifications. Was alright to dismiss.

**Pull request merged**